raft/raft.go

```diff
@@ -1602,6 +1594,29 @@ func (r *raft) abortLeaderTransfer() {
 	r.leadTransferee = None
 }
 
+// CommittedEntryInCurrentTerm return true if the peer has committed an entry in its term.
+func (r *raft) CommittedEntryInCurrentTerm() bool {
+	return r.raftLog.zeroTermOnErrCompacted(r.raftLog.term(r.raftLog.committed)) == r.Term
+}
+
+// responseToReadIndexReq constructs a response for `req`. If `req` comes from the peer
+// itself, a blank value will be returned.
+func (r *raft) responseToReadIndexReq(req pb.Message, committed uint64) pb.Message {
```

Comment:
it should not be `committed`. it is the safe index that the requested node can read. if it is the local node, then then current commit index is OK. if it is from a remote node, the original read index should be preserved.  

probably rename `committed` to `readIndex`